src/models/state/mod.rs

```diff
@@ -1204,17 +1194,33 @@ impl GlobalState {
                     .expect("UTXO notification from miner must be accepted");
             }
 
+            // Get parent of tip for mutator-set data needed for various updates
+            let tip_parent = myself
+                .chain
+                .archival_state()
+                .get_tip_parent()
+                .await
+                .expect("Parent must exist when storing a new block");
+
+            // Sanity check
```

Comment:
Are you referring to the sanity check? That is something I like to add to code that's difficult or has shown itself to be error prone. It makes it easy to identify an error, and it suits my style of rather paranoid programming. It could be argued, though, that any such problem will be/or should be caught by tests instead, not by inlined assert statements.

Reply:
yeah, I was referring to the assert, but my concern is not really with the assert itself, but rather if we are reasonably sure that our calling code prevents this assert from occurring, ie the big picture.  if so, then the assert seems fine.   If not, then either the calling code may need modification, or this fn should return an error rather than asserting.

Reply:
We are reasonably sure the this assert is never triggered. I expanded a bit on comments in `set_new_tip_internal` to make things a bit more clear.